test/test_allocator.cpp

```diff
@@ -25,65 +25,73 @@
 # define CLASSNAME(NAME, SUFFIX) NAME
 #endif
 
+using osrf_testing_tools_cpp::memory_tools::disable_monitoring_in_all_threads;
+using osrf_testing_tools_cpp::memory_tools::enable_monitoring_in_all_threads;
+using osrf_testing_tools_cpp::memory_tools::expect_no_calloc_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_free_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_malloc_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_realloc_begin;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_calloc;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_free;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_malloc;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_realloc;
+
 class CLASSNAME (TestAllocatorFixture, RMW_IMPLEMENTATION) : public ::testing::Test
 {
 public:
-  CLASSNAME(TestAllocatorFixture, RMW_IMPLEMENTATION)() {
-    start_memory_checking();
-    stop_memory_checking();
-  }
+  CLASSNAME(TestAllocatorFixture, RMW_IMPLEMENTATION)() {}
+
   void SetUp()
   {
-    set_on_unexpected_malloc_callback([]() {EXPECT_FALSE(true) << "UNEXPECTED MALLOC";});
-    set_on_unexpected_realloc_callback([]() {EXPECT_FALSE(true) << "UNEXPECTED REALLOC";});
-    set_on_unexpected_free_callback([]() {EXPECT_FALSE(true) << "UNEXPECTED FREE";});
-    start_memory_checking();
+    osrf_testing_tools_cpp::memory_tools::initialize();
+    enable_monitoring_in_all_threads();
   }
 
   void TearDown()
   {
-    assert_no_malloc_end();
-    assert_no_realloc_end();
-    assert_no_free_end();
-    stop_memory_checking();
-    set_on_unexpected_malloc_callback(nullptr);
-    set_on_unexpected_realloc_callback(nullptr);
-    set_on_unexpected_free_callback(nullptr);
+    disable_monitoring_in_all_threads();
+    osrf_testing_tools_cpp::memory_tools::uninitialize();
   }
 };
 
 /* Tests the default allocator.
  */
 TEST_F(CLASSNAME(TestAllocatorFixture, RMW_IMPLEMENTATION), test_default_allocator_normal) {
-  ASSERT_NO_MALLOC(
+  EXPECT_NO_MALLOC(
```

Comment:
is there a default callback? if not, I think that since the callbacks have been removed [above](https://github.com/ros2/rcutils/pull/101/files#diff-f0bc7d22597f044081f4f76be91940d9L37), this won't do anything if an unexpected malloc occurs? Maybe it should be wrapped in [something like this?](https://github.com/osrf/osrf_testing_tools_cpp/blob/88223568ce50cdbe30ab5780154716d9438e83d9/test_osrf_testing_tools_cpp/test/test_example_memory_tools.cpp#L66)

Reply:
Looks like a bug. I should setup the callbacks before that.